File: app/src/main/java/com/akari/ppx/xp/hook/code/purity/TeenHook.java
package com.akari.ppx.xp.hook.code.purity;

import android.app.Activity;

import com.akari.ppx.xp.hook.code.SuperbHook;

import de.robv.android.xposed.XC_MethodReplacement;

import static com.akari.ppx.common.constant.Prefs.REMOVE_TEENAGER_MODE_DIALOG;

public class TeenHook extends SuperbHook {
	@Override
	protected void onHook(ClassLoader cl) {
		hookMethod(REMOVE_TEENAGER_MODE_DIALOG, "com.sup.superb.m_teenager.TeenagerService", "tryShowTeenagerModeDialog", Activity.class, XC_MethodReplacement.DO_NOTHING);
	}
}

File: app/src/main/java/com/akari/ppx/xp/hook/purity/TeenHook.kt
@file:Suppress("unused")

package com.akari.ppx.xp.hook.purity

import android.app.Activity
import com.akari.ppx.utils.replaceMethod
import com.akari.ppx.xp.Init.cl
import com.akari.ppx.xp.hook.SwitchHook

class TeenHook : SwitchHook("remove_teenager_dialog") {
    override fun onHook() {
        "com.sup.superb.m_teenager.TeenagerService".replaceMethod(
            cl,
            "tryShowTeenagerModeDialog",
            Activity::class.java
        ) {}
    }
}